test.py

```diff
@@ -105,6 +105,47 @@ def test_general_case_integers(self):
 
         self.assertTrue(stack.is_empty())
 
+    def test_push_peek_jz(self):
+        stack = StringStack(initial_size=5)
+        self.assertTrue(stack.is_empty())
+        self.assertRaises(SizeError, stack.peek)
+
+        stack.push("Item 1")
+        self.assertEqual(stack.size, 1)
+        self.assertEqual(stack.peek(), "Item 1")
+        self.assertFalse(stack.is_empty())
+
+        stack.push("Item 2")
+        stack.push("Item 3")
+        self.assertEqual(stack.size, 3)
+        self.assertEqual(stack.peek(), "Item 3")
+
+        stack.push("Item 4")
+        stack.push("Item 5")
+        self.assertRaises(SizeError, stack.push, "Item 6")
+
+    def test_pop_jz(self):
+        stack = IntegerStack(initial_size=5)
+        self.assertTrue(stack.is_empty())
+        self.assertRaises(SizeError, stack.pop)
+
+        for i in range(5):
+            stack.push(i)
+
+        self.assertFalse(stack.is_empty())
+
+        self.assertEqual(stack.pop(), 4)
+        self.assertEqual(stack.size, 4)
+        self.assertEqual(stack.peek(), 3)
+
+        stack.push(5)
+        self.assertEqual(stack.peek(), 5)
+        self.assertEqual(stack.pop(), 5)
+
+        for i in range(4):
```

Comment:
Since we're not using `i` here, we can use python good practices to help with the readability:
```
for _ in range(4):
    ...
```